pkg/ebpf/manager.go

```diff
@@ -63,17 +63,17 @@ func NewManagerWithDefault(mgr *manager.Manager, modifiers ...Modifier) *Manager
 type Modifier interface {
 	fmt.Stringer
 	// BeforeInit is called before the ebpf.Manager.InitWithOptions call
-	BeforeInit(*manager.Manager, *manager.Options) error
+	BeforeInit(*manager.Manager, *manager.Options, io.ReaderAt) error
 
 	// AfterInit is called after the ebpf.Manager.InitWithOptions call
-	AfterInit(*manager.Manager, *manager.Options) error
+	AfterInit(*manager.Manager, *manager.Options, io.ReaderAt) error
 }
 
 // InitWithOptions is a wrapper around ebpf-manager.Manager.InitWithOptions
 func (m *Manager) InitWithOptions(bytecode io.ReaderAt, opts *manager.Options) error {
 	for _, mod := range m.EnabledModifiers {
 		log.Debugf("Running %s manager modifier", mod)
-		if err := mod.BeforeInit(m.Manager, opts); err != nil {
+		if err := mod.BeforeInit(m.Manager, opts, bytecode); err != nil {
```

Comment:
I think using `bytecode` as-is will cause problems because it will advance the reader.

Reply:
From the documentation:
`If ReadAt is reading from an input source with a seek offset, ReadAt should not affect nor be affected by the underlying seek offset.`
I interpreted this to mean that `io.ReaderAt` does not modify the curson of the underlying file object.

Reply:
ok, as long as we are using just `io.ReaderAt` it should be OK. I was thinking of `io.Reader`, which the `bytecode.AssetReader` does also implement and would have this problem.